### PR TITLE
SCP-2438 - Add formatting to constant params

### PR DESCRIPTION
**Pull request by palas**
This PR essentially adds the formatting options to value template parameters in the same way as we added it for choices in SCP-2394, SCP-2408, and SCP-2409.

It also adds formatting to the errors when adding choices out of range (which was missing)

Probably easiest to look at the commits independently than the full diff at once

Deployed to: http://pablo.marlowe.iohkdev.io/

Pre-submit checklist:
- Branch
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [x] Relevant tickets are mentioned in commit messages
    - [x] Formatting, materialized Nix files, PNG optimization, etc. are updated
- PR
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
    - [x] Reviewer requested

Pre-merge checklist:
- [x] Someone approved it
- [x] Commits have useful messages
- [x] Review clarifications made it into the code
- [x] History is moderately tidy; or going to squash-merge

### CODE REVIEW

File: marlowe-playground-client/src/MetadataTab/View.purs [hunk 1]
```diff
@@ -41,76 +41,82 @@ onlyDescriptionRenderer setAction deleteAction key info needed metadataAction ty
...
 
-choiceMetadataRenderer :: forall a p. String -> ChoiceInfo -> Boolean -> (MetadataAction -> a) -> String -> String -> Array (HTML p a)
-choiceMetadataRenderer key info@{ choiceFormat } needed metadataAction typeNameTitle typeNameSmall =
+type FormattedNumberActions
+  = { setFormat :: String -> NumberFormat -> MetadataAction
+    , setDescription :: String -> String -> MetadataAction
+    , deleteInfo :: String -> MetadataAction
+    }
+
+formattedNumberMetadataRenderer :: forall a p. String -> String -> NumberFormat -> FormattedNumberActions -> Boolean -> (MetadataAction -> a) -> String -> String -> Array (HTML p a)
```
Comment (hrajchert) [inline marlowe-playground-client/src/MetadataTab/View.purs:50]: we have quite the amount of arguments here and 4 of them are strings. If we are not relying on currying and partial application, I'd suggest to switch to an Object in a future PR and add some comments to the function

File: marlowe-playground-client/src/SimulationPage/View.purs [hunk 2]
```diff
@@ -238,9 +238,20 @@ startSimulationWidget metadata { initialSlot, templateContent } =
...
+  lookupDescription k m =
+    ( case Map.lookup k m of
+        Just { valueParameterDescription: description }
+          | trim description /= "" -> Just description
+        _ -> Nothing
+    )
 
-integerTemplateParameters :: forall action p. Map String String -> (IntegerTemplateType -> String -> BigInteger -> action) -> IntegerTemplateType -> String -> String -> Map String BigInteger -> Array (HTML p action)
-integerTemplateParameters explanations actionGen typeName title prefix content =
+integerTemplateParameters :: forall a action p. (String -> Maybe (String /\ Int)) -> (String -> Map String a -> Maybe String) -> Map String a -> (IntegerTemplateType -> String -> BigInteger -> action) -> IntegerTemplateType -> String -> String -> Map String BigInteger -> Array (HTML p action)
```
Comment (hrajchert) [inline marlowe-playground-client/src/SimulationPage/View.purs:253]: same applies here. I think passing an object with params will make this easier to read

**Review by hrajchert — APPROVED** [review]
LGTM... minor note, with this new formatter, does this description still makes sense? or should it now be "Amount of ADA..."

Screen Shot 2021-07-01 at 10 29 03

**Comment by palas** [comment]
> LGTM... minor note, with this new formatter, does this description still makes sense? or should it now be "Amount of ADA..."
 
I think we probably can remove it once Marlowe Run has the same, since metadata is for the simulator in the first place

**Pull request merged**